internal/net/http/client/client_test.go

```diff
@@ -84,36 +113,73 @@ func TestNew(t *testing.T) {
 		if !errors.Is(err, w.err) {
 			return errors.Errorf("got_error: \"%#v\",\n\t\t\t\twant: \"%#v\"", err, w.err)
 		}
-		if !reflect.DeepEqual(got, w.want) {
-			return errors.Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+		if diff := comparator.Diff(got, w.want, clientComparator...); diff != "" {
+			return errors.New(diff)
 		}
 		return nil
 	}
 	tests := []test{
-		// TODO test cases
-		/*
-		   {
-		       name: "test_case_1",
-		       args: args {
-		           opts: nil,
-		       },
-		       want: want{},
-		       checkFunc: defaultCheckFunc,
-		   },
-		*/
-		// TODO test cases
-		/*
-		   func() test {
-		       return test {
-		           name: "test_case_2",
-		           args: args {
-		           opts: nil,
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+		{
+			name: "initialize with no option success",
+			args: args{
+				opts: nil,
+			},
+			want: want{
+				want: &http.Client{
+					Transport: htr.NewExpBackoff(
+						htr.WithRoundTripper(func() *http.Transport {
+							t := new(http.Transport)
+							t.Proxy = http.ProxyFromEnvironment
+							_ = http2.ConfigureTransport(t)
+
+							return t
+						}()),
+						htr.WithBackoff(
+							backoff.New(),
+						),
+					),
+				},
+			},
+		},
+		{
+			name: "log invalid option error",
```

Comment:
I think it is better.
```suggestion
			name: "fails and log invalid option error",
```